### PR TITLE
Fixes mapstart and random arcade machines

### LINKED ISSUE #53951: Arcade machines with no specified type are not spawning in a new random machine.
From #52970 
https://github.com/tgstation/tgstation/commit/c2b8b5c0cef15e2cd6be4da4a43f1ac91503de31#diff-82e2a70ca3e72867d4a8eec7fce6b1f0
The "New Machine" part was removed, I'm not familiar enough with the code itself to say why, but it's not working as it should.
The circuit works fine, but the name doesn't change and you can't like, play a game on it.
image

### PR DESCRIPTION
**Pull request by Timberpoes**
## About The Pull Request

Fixes #53951

Randomly generated arcade machines were getting random boards and initing properly, but they weren't re-built as the appropriate subtype.

This oversight has been remedied. A new subtype has been created for random arcade machines which exists purely to pick a random circuit then create a machine from that circuit's build path before returning that it should be qdel'd.

This new subtype has been appropriately regex find-and-replaced into all maps.

## Why It's Good For The Game

Pinball. Lizard.

## Changelog
:cl:
fix: Arcade machines no longer require deconstructing and reconstructing in order to work.
/:cl:

### CODE REVIEW

File: code/game/machinery/computer/arcade.dm [hunk 1]
```diff
@@ -142,10 +134,33 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
...
+	name = "random arcade machine"
+	desc = "Automagically transforms into a random arcade machine. If you see this, please create a bug report."
 
-// ** BATTLE ** //
+/obj/machinery/computer/arcade/random/Initialize()
+	// This is just a shell for a random arcade machine. We're not going to do anything
+	// with it, so don't call parent.
+	SHOULD_CALL_PARENT(FALSE)
+
+	var/list/gameodds = list(
```
Comment (Rohesie) [inline code/game/machinery/computer/arcade.dm:146]:
```suggestion
	var/static/list/gameodds = list(
```

File: code/game/machinery/computer/arcade.dm [hunk 2]
```diff
@@ -142,10 +134,33 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
...
 
+/obj/machinery/computer/arcade/random
+	name = "random arcade machine"
+	desc = "Automagically transforms into a random arcade machine. If you see this, please create a bug report."
 
-// ** BATTLE ** //
+/obj/machinery/computer/arcade/random/Initialize()
+	// This is just a shell for a random arcade machine. We're not going to do anything
+	// with it, so don't call parent.
+	SHOULD_CALL_PARENT(FALSE)
```
Comment (Rohesie) [inline code/game/machinery/computer/arcade.dm:144]: This should be a spawner object in that case.

**Comment by Timberpoes** [comment]
Turned into spawner and made list static. Tested locally.

**Review by Rohesie — APPROVED** [review]

**Comment by Timberpoes** [comment]
Opening and closing to clear spurious CI error.

**Comment by Rohesie** [comment]
As a maintainer you can force travis and appveyor restart without having to open and close the PR.

**Comment by LemonInTheDark** [comment]
Why did you remove the behavior from the subtype in the first place? cleanliness?

**Comment by Timberpoes** [comment]
> 
> 
> Why did you remove the behavior from the subtype in the first place? cleanliness?

Was hoping to clear up a snowflake machine in general that Inited itself just to destroy itself from Init.

The spawner solution is much, much cleaner though.

**Pull request merged**